### PR TITLE
Restructure dependency page for better readability

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
**What this PR does / why we need it**:
- Restructure dependency page for better readability

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Restructure dependency page for better readability
```

### CODE REVIEW

File: docs/book/src/developer/dependencies.md [hunk 1]
```diff
@@ -1,26 +1,46 @@
...
+| Package | Module name | Used by |
+| ------- | ----------- | ------- |
 | cluster-api | [sigs.k8s.io/cluster-api](https://github.com/kubernetes-sigs/cluster-api) | [go.mod][go.mod1] |
 | cluster-api/test | [sigs.k8s.io/cluster-api/test](https://github.com/kubernetes-sigs/cluster-api/tree/main/test) | [go.mod][go.mod1]  |
 | cluster-api/hack/tools | [sigs.k8s.io/cluster-api/hack/tools](https://github.com/kubernetes-sigs/cluster-api/tree/main/hack/tools) | [hack/tools/go.mod][go.mod2] |
-| controller-runtime | [sigs.k8s.io/controller-runtime](https://sigs.k8s.io/controller-runtime) | [go.mod][go.mod1] |
-| controller-runtime/tools/setup-envtest | [sigs.k8s.io/controller-runtime/tools/setup-envtest](https://sigs.k8s.io/controller-runtime/tools/setup-envtest) | [hack/tools/go.mod][go.mod2] |
-| controller-tools | [sigs.k8s.io/controller-tools](https://sigs.k8s.io/controller-tools) | [hack/tools/go.mod][go.mod2] |
+
+#### IBM Packages
```
Comment (mkumatag) [inline docs/book/src/developer/dependencies.md:10]: I think except these ibm packages, other packages are updated with every capi version update right? so wondering if we can bring that message here

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: docs/book/src/developer/dependencies.md (hunk 1 above)
Comment (Prajyot-Parab) [reply docs/book/src/developer/dependencies.md:10]: Yes will add that note.

Comment (Prajyot-Parab) [reply docs/book/src/developer/dependencies.md:12]: Done.

File: docs/book/src/developer/dependencies.md [hunk 2]
```diff
@@ -1,26 +1,48 @@
...
 | cli-runtime | [k8s.io/cli-runtime](https://k8s.io/cli-runtime) | [go.mod][go.mod1] |
 | client-go | [k8s.io/client-go](https://k8s.io/client-go) | [go.mod][go.mod1] |
 | utils | [k8s.io/utils](https://k8s.io/utils) | [go.mod][go.mod1] |
 
+#### Controller Packages
+| Package | Module name | Used by |
+| ------- | ----------- | ------- |
+| controller-runtime | [sigs.k8s.io/controller-runtime](https://sigs.k8s.io/controller-runtime) | [go.mod][go.mod1] |
+| controller-runtime/tools/setup-envtest | [sigs.k8s.io/controller-runtime/tools/setup-envtest](https://sigs.k8s.io/controller-runtime/tools/setup-envtest) | [hack/tools/go.mod][go.mod2] |
+| controller-tools | [sigs.k8s.io/controller-tools](https://sigs.k8s.io/controller-tools) | [hack/tools/go.mod][go.mod2] |
```
Comment (mkumatag) [inline docs/book/src/developer/dependencies.md:35]: these things can be clubbed into one and moved under subsection of capi packages

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: docs/book/src/developer/dependencies.md (hunk 2 above)
Comment (Prajyot-Parab) [reply docs/book/src/developer/dependencies.md]: Done.

Comment (Prajyot-Parab) [reply docs/book/src/developer/dependencies.md]: Done.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**